src/middlewares/permissions.js

```diff
@@ -86,6 +88,50 @@ class PermissionMiddlewareCreator {
     };
   }
 
+  static _getRequestAttributes(request) {
+    const hasBodyAttributes = request.body && request.body.data && request.body.data.attributes;
+    return hasBodyAttributes
+      && new QueryDeserializer(request.body.data.attributes).perform();
+  }
+
+  // generate a middleware that will check that ids provided by the request exist
+  // whithin the registered scope
+  _ensureRecordIdsInScope(model) {
+    if (!model) throw new Error('missing model');
+
+    return async (request, response, next) => {
+      const attributes = PermissionMiddlewareCreator._getRequestAttributes(request);
+
+      // if performing a `selectAll` let the `getIdsFromRequest` handle the scopes
+      if (attributes.allRecords) return next();
+
+      const modelName = this.configStore.Implementation.getModelName(model);
+      const { idField } = Schemas.schemas[modelName];
+
+      // TODO: scope smartAction calls properly on table with composite primary keys
+      if (idField === 'forestCompositePrimary') return next();
```

Comment:
Maybe we should debate with Arnaud if it's OK to allow this.
The use case is B2B2B, security matters much more than with all previous forest admin features.

If two customers which don't know each other are using forest admin, we really don't want information to leak between them. 

How about:
```
return response.status(400).send({ error: 'Smart Action are not supported on models with composite primary keys' });
```

Reply:
Just to be clear, the code is OK.
It's just that I think that it is better if arnaud knows about the security issue before we merge this.

This way, he won't be surprised if the next security audit finds it

Reply:
The idea here was to have a first version working for non-compositeKeys collection (as we're targeting a mongoose project first) and later on the project add the support for that